File: turi/src/main/java/com/turi/offer/domain/model/Offer.java
package com.turi.offer.domain.model;

import com.turi.attraction.domain.model.Attraction;
import com.turi.stay.domain.model.StayDto;
import com.turi.touristicplace.domain.model.GuaranteedService;
import com.turi.touristicplace.domain.model.TouristicPlace;
import lombok.*;

import java.util.List;

@Getter
@Setter
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
@Builder(setterPrefix = "with")
public final class Offer
{
    private TouristicPlace touristicPlace;
    private List<GuaranteedService> guaranteedServices;
    private List<StayDto> stays;
    private List<Attraction> attractions;
}

File: turi-ui/src/types/offer.ts
import { GuaranteedService, TouristicPlace } from './touristicPlace.ts'
import { StayDto } from './stay.ts'
import { Attraction } from './attraction.ts'

export interface Search {
    offers: Offer[]
    touristicPlaceId: number
    rank: number
}

export interface Offer {
    touristicPlace: TouristicPlace
    guaranteedServices: GuaranteedService[]
    stays: StayDto[]
    attractions: Attraction[]
}

export enum SearchMode {
    ALL = 'ALL',
    STAY = 'STAY',
    ATTRACTION = 'ATTRACTION'
}
